SIL.Archiving/ArchivingDlgViewModel.cs

```diff
@@ -546,6 +546,29 @@ public static bool IsMono
 
 		/// <remarks/>
 		public Dictionary<string, MessageType> AdditionalMessages { get; private set; }
+
```

Comment:
Let's put this in the DirectoryUtilities class and call it IsDirectoryWritable. Rather than creating/overwriting a file that could be present, generate a sub-folder name (ensure uniqueness by calling GetUniqueFolderPath) and just attempt to create (and then clean up) the subfolder. If we can create a subfolder, we should be assured of full write access.

Reply:
`GetUniqueFolderPath(folderPath)` returns a directory name that is in the parent of `folderPath`, so I'm instead I'm using
```csharp
var tempPath = Path.Combine(folderPath, Path.GetRandomFileName());
```

Reply:
Done.